api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -201,6 +201,51 @@
+     * {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}. 
+     * If the type is annotated with {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}, 
+     * the prefix supplied in this method overrides the prefix associated with the annotation 
+     * {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}.
+     * 
+     * @param <T> 
+     *              The Class Type
+     * @param configProperties
+     *              The class that contains a number of fields that maps to corresponding configuration properties. 
+     *              The type declaration can be annotated with 
+     *              {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}.
+     *              The prefix as the method parameter overrides the prefix set on the type.                
+     *              This class must contain a zero-arg constructor.
+     * @param prefix
+     *              The prefix for the configuration properties declared on the class configProperties.
+     *              If the prefix is "", which means no prefix involved when performing property lookup.
+     *              If the prefix is null, this method is equivalent to {@linkplain #getConfigProperties(Class)}.
+     * @return      An instance for the specified type and prefix 
+     */
+    <T> T getConfigProperties(Class<T> configProperties, String prefix);
+
+    /**
+     * Return the resolved configuration properties instance. The type can be annotated with 
+     * {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}. 
+     * If the type is annotated with {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}, 
+     * the prefix associated with the annotation {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties} will be used.
+     * @param <T> 
+     *              The Class Type
+     * @param configProperties
+     *              The class that contains a number of fields that maps to corresponding configuration properties. 
+     *              The type declaration can be annotated with 
+     *              {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}. 
+     *              If the type is annotated with {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}, 
+     *              the prefix on the annotation {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties} 
+     *              will be honoured. 
+     *              The absence of the annotation {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties} 
+     *              or the absence of the prefix on the annotation means no prefix specified. 
+     *              This class must contain a zero-arg constructor.
```

Comment:
Thank you @OndroMih for re-reviewing this! Very much appreciated! As for the zero-arg requirement, I thought this further and agreed it is required for creating a proxy. However, for programmatic lookup, this can be relaxed. 

I am ok to update to "This class should contain a zero-arg constructor, otherwise the behaviour is unspecified." and delete the corresponding test.

Reply:
I'm ok with this.

Reply:
updated.

Reply:
I thought this further:
Instead of saying 

This class should contain a zero-arg constructor. Otherwise, the behaviour is unspecified.

should we say:
**This class should contain a zero-arg constructor. Otherwise, non-portable behaviour occurs.**

as this phrase "non-portable" is common in specs?

@OndroMih @jbee @radcortez thoughts?

Reply:
updated accordingly.